ingestion_server/ingestion_server/indexer.py

```diff
@@ -469,3 +411,60 @@ def point_alias(self, model_name: str, index_suffix: str, alias: str, **_):
         if self.progress is not None:
             self.progress.value = 100  # mark job as completed
         self.ping_callback()
+
+    def delete_index(
+        self,
+        model_name: str,
+        index_suffix: Optional[str] = None,
+        alias: Optional[str] = None,
+        force_delete: bool = False,
+        **_,
+    ):
+        """
+        Delete the given index ensuring that it is not in use.
+
+        :param model_name: the name of the media type
+        :param index_suffix: the suffix of the index to delete
+        :param alias: the alias to delete, including the index it points to
+        :param force_delete: whether to delete the index even if it is in use
+        """
+
+        target = alias if alias is not None else f"{model_name}-{index_suffix}"
+
+        target_stat = get_stat(self.es, target)
+        if target_stat.exists:
+            if target_stat.is_alias:
+                if not force_delete:
+                    # Alias cannot be deleted unless forced.
+                    message = (
+                        f"Alias {target} might be in use so it cannot be deleted. Use"
+                        f"the `force_delete` parameter."
+                    )
+                    log.error(message)
+                    slack.error(message)
+                    return
+                target = target_stat.alt_names
+            else:
+                if target_stat.alt_names:
+                    # Index mapped to alias cannot be deleted.
+                    message = (
+                        f"Index {target} is associated with aliases "
+                        f"{target_stat.alt_names}, cannot delete. Delete aliases first."
+                    )
+                    log.error(message)
+                    slack.error(message)
+                    return
+
+            self.es.indices.delete(index=target)
+            message = f"Index {target} was deleted."
+            log.info(message)
+            slack.info(message)
+        else:
+            # Cannot delete as target does not exist.
+            message = f"Target {target} does not exist and cannot be deleted."
+            log.info(message)
+            slack.info(message)
```

Comment:
Should this generate a bad request response?

Reply:
The issue with that is that the ingestion server registers requests, returns the response and then processes tasks in a separate process. So unless the task is executed, every request with a valid schema is a good request.

Reply:
Tasks with invalid input will eventually fail so the task status endpoint can be used after submitting to see if the task passes (and completes) or fails.

Reply:
I'll rescind the above statements, I've added a framework 72cc511 for tasks to communicate bad requests to the main API process, at least for the failing-fast cases where it's immediately possible to determine if the input is invalid.